js/tji-datasets-explore.js

```diff
@@ -1,15 +1,349 @@
+  this.filters_elt_id = filters_elt_id;
+  this.chart_wrapper = chart_wrapper;
+  this.count_template = count_template;
+  this.filters = null;
+
+  this.get_data();
+}
+
+ChartView.prototype.missing_data_label = '(not given)';
+
+ChartView.prototype.get_data = function() {
+  var that = this;
+  jQuery.getJSON('/cleaned_custodial_death_reports.json')
+    .done(function(data){
+      that.state.data = data;
+      that.parse_data();
+      that.create_charts();
+      that.create_filter_panel();
+      that.attach_events();
+    })
+    .fail(function(e){
+      console.log('error fetching data from: ' + this.state.url, e);
+    })
+}
+
+ChartView.prototype.parse_data = function() {
+
+  var that = this;
+  var column_whitelist = [
+    'year', 'race', 'sex', 'manner_of_death', 'age_group',
+    'type_of_custody', 'death_location_type', 'means_of_death',
+  ]
+
+    // Create new columns
+
+  _.each(this.state.data, function(data_row, i) {
+    // Build column for year
+    data_row['year'] = data_row['death_date'].substring(0, 4);
+
+    // Create age group buckets
+    if (data_row['age_at_time_of_death'] < 0) {
+      data_row['age_group'] = null;
+    } else {
+      age_decade = Math.floor(data_row['age_at_time_of_death'] / 10) * 10
+      if (age_decade > 59) {
+        data_row['age_group'] = '60+'
+      } else {
+        data_row['age_group'] = age_decade + '-' + (age_decade + 9)
+      }
+    }
+
+    _.each(column_whitelist, function(key) {
+      if (data_row[key] === undefined || data_row[key] === null || data_row[key] === '') {
+        data_row[key] = that.missing_data_label;
+      }
+    });
+  });
+
+  // Generate filter values
+
+  filters = []
+  _.each(column_whitelist, function(column) {
+    // Get a sorted list of all the unique values for this column
+    values = _.filter(_.sortBy(_.uniq(_.map(that.state.data, column))), _.identity);
+    if (values.indexOf(that.missing_data_label) !== -1) {
+      values.splice(values.indexOf(that.missing_data_label), 1);
+      values.push(that.missing_data_label);
+    }
+    filters.push({
+      key: column,
+      values: values
+    });
+  });
+  this.filters = filters;
+}
+
+ChartView.prototype.create_filter_panel = function() {
+  var $filters = jQuery('<form id="js-TJIfilters" />');
+  _.each(this.filters, function(f) {
+    var fieldset = jQuery('<fieldset><legend>' + f.key.replace(/_/g, " ") + '</legend></fieldset>');
+    _.each(f.values, function(v) {
+      var input = jQuery('<input/>', {
+        type: "checkbox",
+        checked: "checked",
+        id: v,
+        name: f.key,
+        value: v,
+      });
+      var label = jQuery('<label/>', {
+        for: v,
+      }).text(v);
+      fieldset.append(jQuery('<div></div>').append(input, label));
+    });
+    $filters.append(fieldset);
+  });
+  jQuery(this.filters_elt_id).append($filters);
+  this.state.active_filters = jQuery(this).serializeArray();  
+}
+
+ChartView.prototype.create_charts = function() {
```

Comment:
Maybe this should be TJIChartView? I know I named it, but I was thinking you'd want more work Sunday night.

Reply:
FINE
![table_flip](https://user-images.githubusercontent.com/1421476/41822212-d7ade662-77b1-11e8-99b8-f40b8597c966.gif)